Product addon beta banner

ThomazFB opened the pull request:
Summary
==========
Fixes issue #4422 by creating the Beta banner for the Ordered Addons view with the ability to redirect the merchant to the Survey form and to dismiss the banner itself.

Screenshots
==========

## Figma Designs
![image](https://user-images.githubusercontent.com/5920403/130272949-153ff241-0f27-4f0f-9aa6-b4138255e7de.png)

## Result
| Ordered Addons with banner collapsed  | Ordered Addons with banner expanded |
| ------------- | ------------- |
| ![Screenshot_20210820_020207](https://user-images.githubusercontent.com/5920403/130272856-d235bf8b-bfb8-403d-b373-9a8aca7e8986.png) | ![Screenshot_20210820_020210](https://user-images.githubusercontent.com/5920403/130272862-ad294800-1ed7-4f92-9a82-d2bc77dd94b0.png) |

How to Test
==========
Prerequisites: Have your site with the add ons plugin installed, configured, and with at least one product with add-ons and at least one order of the configured product. ⚠️ Note that you should configure the Product directly from the Edit Product page of wp-admin, if you create Global add-ons only, they will not show up here yet.

1 - Launch the Woo app.
2 - Navigate to an order with a Product configured with Add-ons, verify if the `View Add-ons` Button is showing up for products that contain Add-ons inside the current Order, but doesn't show up for Products without Add-ons for that order or Products that don't have Add-ons configured at all.
3 - Click on `View Add-ons` and verify if the Ordered Addons view is displayed with all the expected data.
4 - Interact with the feature banner, verify if it's collapsing and expanding as expected and if each action is working as expected, including the Feedback Survey form presentation.


Update release notes:

- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ThomazFB committed:
Suppress Static field leak lint warning

ThomazFB committed:
Add Feature banner element to fragment_ordered_addon

ThomazFB committed:
Add Product Addon feature to FeatureFeedbackSettings

ThomazFB committed:
Add Feature state handling to OrderedAddonFragment

ThomazFB committed:
Improve FeatureFeedbackSettings handling functions

ThomazFB committed:
Create Feature banner presentation logic

ThomazFB committed:
Adjust Feedback setting structure

ThomazFB committed:
Extract WIP banner text to strings.xml

ThomazFB committed:
Fix lint issues

ThomazFB committed:
Refactor view initialization at OrderedAddonFragment

ThomazFB committed:
Fix lint issues

ThomazFB committed:
Add selectable background to View Add-ons button

ThomazFB committed:
MOve actionbar setup to setupViews method at OrderedAddonFragment

nbradbury commented:
@ThomazFB If I expand the WIP card then tap to give feedback, when I return from giving feedback the card is collapsed and I have to tap twice to expand it. The same thing happens if I rotate the device.

nbradbury commented:
@ThomazFB If I tap to dismiss the card, the next time I view add-ons the card still appears.

ThomazFB committed:
Manually set Beta banner to collapsed whenever configuration changes

ThomazFB committed:
Fix OrderedAddonFragment state control for Beta Banner

ThomazFB commented:
Hey @nbradbury, thanks for reviewing the PR! I've submitted some fixes, and the PR is ready for another round 😄 

>when I return from giving feedback, the card is collapsed, and I have to tap twice to expand it. The same thing happens if I rotate the device.

I've fixed this problem in this commit right here: https://github.com/woocommerce/woocommerce-android/pull/4654/commits/f52088507e4fe7ba58c3023af8e08d3cee09e90b. But I decided to fix the double-tap issue only. The reason is that keeping the collapsed state after rotating or exiting the View would require us to add a good amount of ViewState code control at the ViewModel, and I wasn't sure it would be worth doing so just to maintain the banner expanded when the View is reconfigured. WDYT about this? Do you believe we should control it anyway?

By the way, I placed the `isExpanded` from the `FeatureWIPNoticeCard` as a public field since, from my perspective, this is a View state control that the Fragment should also be able to do to define.

>If I tap to dismiss the card, the card still appears the next time I view add-ons.

Fixed here: https://github.com/woocommerce/woocommerce-android/pull/4654/commits/5f0052b906f2e5b2b95e38491b6841d32aeb8703. I forgot to update the dismiss action to place the correct state, thanks for catching this one 😅

nbradbury commented:
> I wasn't sure it would be worth doing so just to maintain the banner expanded when the View is reconfigured. WDYT about this? Do you believe we should control it anyway?

I don't think it's a big deal at all to collapse it. The double tap to me was the real issue, and that's resolved now.

nbradbury reviewed (approved):
Looks good! :shipit:
